src/main.py

```
@@ -45,20 +50,50 @@ def run_simulate(sim_mode: str) -> None:
+    budget:
+        Total marketing budget in KRW. Defaults to 50,000,000 if not provided.
+    """
+    from optimization.budget import run_budget_pipeline
+
+    try:
+        run_budget_pipeline(
+            data_dir="results",
+            output_dir="results",
+            budget=budget or 50_000_000,
+        )
+    except Exception as exc:
```

Ruff (0.15.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)